tests/cnf/ran/powermanagement/tests/cpufreq.go

```diff
@@ -0,0 +1,100 @@
+package tests
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/eco-goinfra/pkg/nodes"
+	"github.com/openshift-kni/eco-goinfra/pkg/nto" //nolint:misspell
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/cluster"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/raninittools"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/powermanagement/internal/helper"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/powermanagement/internal/tsparams"
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+)
+
+var _ = Describe("Per-core runtime power states tuning", Label(tsparams.LabelCPUFrequency), func() {
+	var (
+		nodeList                []*nodes.Builder
+		perfProfile             *nto.Builder
+		err                     error
+		desiredReservedCoreFreq = performancev2.CPUfrequency(2500002)
+		desiredIsolatedCoreFreq = performancev2.CPUfrequency(2200002)
+		originalIsolatedCPUFreq performancev2.CPUfrequency
+		originalReservedCPUFreq performancev2.CPUfrequency
+		isolatedCPUNumber       = 2
+		ReservedCPUNumber       = 0
+	)
+
+	BeforeEach(func() {
```

Comment:
The BeforeEach instead of BeforeAll was my suggestion since there's only one test in the file the ordered is a bit redundant but if the BeforeEach is expensive then it'd make sense to use continue on failure. My understanding is there's no reboot in this test though

Reply:
@klaskosk right, but we do revert the config in AfterXX, so best to do that in AfterAll I think if we will have multiple  test cases in the future. But we can also change this structure when we actually have more tests in the future. @josclark42  feel free to keep Before/AfterEach for this PR if you would prefer that way.